src/scripts/exclusion_term_expansion.py

```diff
@@ -51,19 +51,29 @@ class having `True` for `exclude_children`, but the child class has `False`.
 
 
 # Functions
-# TODO: use: https://bioregistry.readthedocs.io/
-#  en/latest/api/bioregistry.curie_from_iri.html?highlight=get_curie#bioregistry.curie_from_iri
-def uri_to_curie(uri: str) -> str:
-    """Takes an ontological URI and returns a CURIE. Works on the following patterns:
-    - http://.../PREFIX/CODE
-    - http://.../PREFIX_CODE"""
-    uri_list: List[str] = uri.split('/')
-    end_bit = uri_list[-1]
-    if '_' in end_bit:
-        curie = end_bit.replace('_', ':')
+# TODO: evaluate new bioregistry option https://github.com/biopragmatics/bioregistry/issues/480#issuecomment-1199235747
+def uri_to_curie(uri: str, prefix_map: Dict[str, str] = None) -> str:
+    """Takes an ontological URI and returns a CURIE. Works on the following patterns
+    # todo: defaulting prefix_map to None is temp for now but need to change to oak anyway soon
+    # todo: Use this solution until I get OAK to work for this use case"""
+    obo_purl = "http://purl.obolibrary.org/obo/"
+    curie = uri
+    uri = curie.replace('<', '').replace('>', '')
+    if prefix_map:
+        for k, v in prefix_map.items():
+            if uri.startswith(v):
+                curie = uri.replace(v, f"{k}:")
+        if uri.startswith(obo_purl):
```

Comment:
why is this in the `if prefix_map` section of the code? It should be in the `else` part! If there is a prefix_map, that should always be preferred. Indeed, see comment below, I would prefer if we always use prefix_map, and raise an exception of the URI cannot be checked with the prefix_map. (prefix_map should be required).

Reply:
I saw your comment [on Slack](https://monarchinitiative.slack.com/archives/C0222P0736F/p1659356271536479) about this as well. This was towards the end of a really looong day of work. I was just trying to get one of the libraries to work and clock out for the night. This code came from OAK, actually: https://github.com/INCATools/ontology-access-kit/blob/main/src/oaklib/interfaces/basic_ontology_interface.py#L136

Although there is a `TODO` in there to change that.

It's NP; I understand your logic completely. I'll remove it.

Reply:
done!